daemon/cluster/noderunner.go

```diff
@@ -357,8 +357,10 @@ func (n *nodeRunner) enableReconnectWatcher() {
 	n.cancelReconnect = cancel
 
 	go func() {
+		type timeoutError interface{ Timeout() bool }
+
 		<-delayCtx.Done()
-		if delayCtx.Err() != context.DeadlineExceeded {
+		if _, ok := delayCtx.Err().(timeoutError); !ok {
```

Comment:
Same as previous comment, do we need to also check `DeadlineExceeded` here to keep the behavior the same? Even if no context returns it right now, one could leak in later that doesn't return a timeoutError and the user passing in a context might still expect it to cancel correctly.

Reply:
`DeadlineExceeded` implements `timeoutError`

Reply:
Ah yes, that explains why net.Error isn't used as well. Thanks.